si4432 compile error on MingGW

# Problem
When compiling on my MinGW environment, I get this error:
~~~
C+ Libraries/si4432/si4432.cpp
In file included from Wiring/../include/user_config.h:27:0,
                 from Wiring/Arduino.h:6,
                 from Libraries/si4432/si4432.h:30,
                 from Libraries/si4432/si4432.cpp:27:
Libraries/si4432/si4432.cpp: In member function 'void Si4432::readAll()':
Libraries/si4432/si4432.cpp:400:98: error: iteration 7u invokes undefined behavior [-Werror=aggressive-loop-optimizations]
     (int ) allValues[i+12], (int ) allValues[i+13], (int ) allValues[i+14], (int ) allValues[i+15]
                                                                                                  ^
system/include/esp_systemapi.h:32:48: note: in definition of macro 'debugf'
 #define debugf(fmt, ...) m_printf(fmt"\r\n", ##__VA_ARGS__)
                                                ^
Libraries/si4432/si4432.cpp:394:21: note: containing loop
  for (byte i = 0; i < 0x7f; i+=16)
                     ^
cc1plus.exe: all warnings being treated as errors
make: *** [out/build/Libraries/si4432//si4432.o] Error 1
~~~

# Analysis

Code review shows, that indeed the last loop the last byte for debug output is taken from moutside the allocated buffer.
Since I it happens with stock settings (master+develop in Sming, Toolchain from cholatey install, I assume that this problem could happen to others.

# Environment
 gcc taken from Makefile database:
~~~
/c/Espressif/xtensa-lx106-elf/bin/xtensa-lx106-elf-gcc -v
Using built-in specs.
COLLECT_GCC=c:\Espressif\xtensa-lx106-elf\bin\xtensa-lx106-elf-gcc.exe
COLLECT_LTO_WRAPPER=c:/espressif/xtensa-lx106-elf/bin/../libexec/gcc/xtensa-lx106-elf/5.1.0/lto-wrapper.exe
Target: xtensa-lx106-elf
Configured with: ../configure --prefix=/d/Neo/esp8266-devkit/Espressif/xtensa-lx106-elf --target=xtensa-lx106-elf --enable-multilib --disable-nls --disable-shared --disable-threads --with-gnu-as --with-gnu-ld --with-gmp=/d/Neo/esp8266-devkit/Espressif/build/gmp --with-mpfr=/d/Neo/esp8266-devkit/Espressif/build/mpfr --with-mpc=/d/Neo/esp8266-devkit/Espressif/build/mpc --enable-languages=c,c++ --with-newlib --disable-libssp --disable-__cxa_atexit
Thread model: single
gcc version 5.1.0 (GCC)
~~

--- a/Sming/Libraries/si4432/si4432.cpp
+++ b/Sming/Libraries/si4432/si4432.cpp
@@ -391,7 +391,7 @@ void Si4432::readAll() {
 
 	debugf("REGS  00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F");
 
-	for (unsigned int i = 0; i < 0x7f; i+=16)
+	for (byte i = 0; i < 0x7f; i+=16)
 	{
 		debugf("(%02x): %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x", i,
 				(int ) allValues[i+0], (int ) allValues[i+1], (int ) allValues[i+2], (int ) allValues[i+3],

--- a/Sming/Libraries/si4432/si4432.cpp
+++ b/Sming/Libraries/si4432/si4432.cpp
@@ -385,13 +385,13 @@ void Si4432::BurstRead(Registers startReg, byte value[], uint8_t length) {
 
 void Si4432::readAll() {
 
-	byte allValues[0x7F];
+	byte allValues[0x80];
 
 	BurstRead(REG_DEV_TYPE, allValues, 0x7F);
 
 	debugf("REGS  00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F");
 
-	for (byte i = 0; i < 0x7f; i+=16)
+	for (unsigned int i = 0; i < 0x7f; i+=16)
 	{
 		debugf("(%02x): %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x", i,
 				(int ) allValues[i+0], (int ) allValues[i+1], (int ) allValues[i+2], (int ) allValues[i+3],

